sqlmesh/core/model/definition.py

```diff
@@ -2082,11 +2082,15 @@ def _create_model(
             physical_schema_override = override_schema
             break
 
+    defaults = {k: v for k, v in (defaults or {}).items() if k in klass.all_fields()}
+
     raw_kind = kwargs.pop("kind", None)
     if raw_kind:
-        kwargs["kind"] = create_model_kind(raw_kind, dialect, defaults or {})
-
-    defaults = {k: v for k, v in (defaults or {}).items() if k in klass.all_fields()}
+        parsed_kind = create_model_kind(raw_kind, dialect, defaults or {})
+        kwargs["kind"] = parsed_kind
+        if parsed_kind.is_external or parsed_kind.is_seed or issubclass(klass, PythonModel):
```

Comment:
Should this be `if klass is not SqlModel` or something?

Reply:
Ah yeah maybe so, can simplify by not relying on kind at all I suppose, 1 sec

Reply:
hey is this the best place to put this?, what about in the validator or something

Reply:
Yeah @tobymao , by the time you are in the context of a validator -- the defaults have already been mixed in with explicit arguments. We can see it happening just below my change
https://github.com/z3z1ma/sqlmesh/blob/927d4cba9d87f219e543f737fc470e257dd33ec6/sqlmesh/core/model/definition.py#L2120-L2130

The behavior my change gives us (as can be seen in the unmutated test cases above my new one) is that we still raise if a user explicitly adds `validate_query TRUE` to a SEED or EXTERNAL model in the definition but do NOT raise by injecting a default since we filter it appropriately before it reaches the validators.  

Also latest change is nice and doesn't even look at kind, its more correct now I think and a simple change